### PR TITLE
core/rawdb: freezer batch write

### PR DESCRIPTION
**Pull request by holiman**
This is the rebased and squashed version of https://github.com/ethereum/go-ethereum/pull/22932. Due to the new batched reader, it became pretty difficult to rebase and retain all commits, so I squashed it, which made the whole thing a lot easier. 

@fjl you're welcome to do do a more thorough rebase if you want to keep the commit history, or you can rebase and reown it, I don't really care personally.

### CODE REVIEW

**Comment by holiman** [comment]
I'm not sure how it happened that I'm listed as the only committer on that squash. I thought it would be a mix icon showing both of us. That's a bit bungled...

**Comment by fjl** [comment]
Don't worry about that, we can put me back in the final commit using `Co-authored-by:`.

File: core/blockchain.go [hunk 1]
```diff
@@ -1085,38 +1084,6 @@ const (
...
-		return nil
-	}
-	// Truncate all the data in the freezer beyond the specified head
-	if err := bc.db.TruncateAncients(head + 1); err != nil {
-		return err
-	}
-	// Clear out any stale content from the caches
-	bc.hc.headerCache.Purge()
-	bc.hc.tdCache.Purge()
-	bc.hc.numberCache.Purge()
```
Comment (holiman) [inline core/blockchain.go:1106]: @fjl these are now moved into headerchain, inside `SetHead`

Comment (fjl) [reply core/blockchain.go:1106]: Yes, but we don't call it in all cases.

**Pull request merged**